### ISSUE TITLE
Serializable tween parameters data

### ISSUE THREAD

**Issue by rfadeev** [Demigiant/dotween#186]
Hello,

Have you ever considered wrapping tween parameters data into `Serializable` classes to allow inspector configuration for the tweens? Something like this:

```code chsharp
using System;
using UnityEngine;
using DG.Tweening;

[Serializable]
public class MoveTweenConfig
{
    [SerializeField] private Vector3 m_to;
    [SerializeField] private float m_duration;
    [SerializeField] private bool m_snapping = false;

    public Vector3 to { get { return m_to; }
    public float duration { get { return m_duration; }
    public bool snapping { get { return m_snapping; }
}

public static class DOTweenConfigExtensions
{
    public static Tweener DOMove(this Transform target, MoveTweenConfig c)
    {
        return target.DOMove(c.to, c.duration, c.snapping);
    }
}
```

This would ease several tweens configuration via editor, for example:
```code chsharp
using UnityEngine;

public class TweenConfigExample : MonoBehaviour
{
    [SerializeField] private GameObject targetA;
    [SerializeField] private MoveTweenConfig targetAMoveConfig;
    [SerializeField] private GameObject targetB;
    [SerializeField] private MoveTweenConfig targetBMoveConfig;

    private void Start()
    {
        targetA.transform.DOMove(targetAMoveConfig);
        targetB.transform.DOMove(targetBMoveConfig);
    }
}
```

I was thinking of implementing such wrappers but not sure whether they should be separate repository or part of DOTween.

Regards,
Roman

**Comment by rfadeev** [comment]
For those who might be interested in serializable tween parameters data please check my implementation of this idea: https://github.com/rfadeev/dotween-configs

**Comment by jaidhyani** [comment]
@rfadeev I was just looking for something like this. It looks like the code hasn't been touched in a while, was wondering if you thought it'd be worth trying in a new project or if it's likely to have gotten too out of sync with DOTween development.

**Comment by rfadeev** [comment]
@jaidhyani Hi! Thanks for the interest to the project 🙂

I just tested my project with DOTween v1.2.632 (August 3, 2021) - latest one available at [http://dotween.demigiant.com/download.php](http://dotween.demigiant.com/download.php). I imported DOTween first and then added latest code of DOTween Configs. No compilation errors. Serialized DOTween config worked for a sample script from README, game object moved according to the config values in Unity Editor playmode. I also made standalone Windows build for a the same sample scene and the object moved in the build too.

Code I used:
```csharp
using UnityEngine;
using DOTweenConfigs;

public class DOMoveSerializableExample : MonoBehaviour
{
    [SerializeField]
    private Position3DTweenConfig tweenConfig;

    private void Start()
    {
        transform.DOMove(tweenConfig);
    }
}
```
So if you have ideas or needs - feel free to submit pull request or open an issue, I could take a look to implement what is missing in the project. You are also welcome to inspire from the code there and start your own new project.

Meanwhile, I am going to add Unity package support to ease usage of the project.

Cheers,
Roman
